src/main/java/org/cryptomator/common/vaults/Vault.java

```diff
@@ -115,11 +115,11 @@ private CryptoFileSystem createCryptoFileSystem(MasterkeyLoader keyLoader) throw
 				int cleartextLimit = checker.determineSupportedCleartextFileNameLength(getPath());
 				vaultSettings.maxCleartextFilenameLength().set(cleartextLimit);
 			} else {
-				vaultSettings.maxCleartextFilenameLength().setValue(UNLIMITED_FILENAME_LENGTH);
+				vaultSettings.maxCleartextFilenameLength().setValue(Constants.UNLIMITED_CLEARTEXT_FILENAME_LENGTH);
 			}
 		}
 
-		if (vaultSettings.maxCleartextFilenameLength().get() < UNLIMITED_FILENAME_LENGTH) {
+		if (vaultSettings.maxCleartextFilenameLength().get() < Constants.UNLIMITED_CLEARTEXT_FILENAME_LENGTH) {
 			LOG.warn("Limiting cleartext filename length on this device to {}.", vaultSettings.maxCleartextFilenameLength().get());
 		}
```

Comment:
There is no check preventing the user to enter a limit that exceeds the actually supported length, which will lead to exactly those problems, that the limit tries to prevent.